src/form/Form.tsx

```diff
@@ -1348,6 +1349,10 @@ function Form({
         client,
         integrations
       });
+    } else if (link === LINK_SAVE_HIDDEN_FIELD) {
+      const { hidden_field_id: id, hidden_field_value: value } =
+        button.properties;
+      updateFieldValues({ [id]: value });
```

Comment:
in the submit flow, you'll need to also look through buttons for fields like these whose values might have been updated that we want to send back to the backend. you can send hidden field values via submitCustom